cmd/debos/debos.go

```diff
@@ -15,18 +15,18 @@ import (
 	"github.com/jessevdk/go-flags"
 )
 
-func checkError(context *debos.DebosContext, err error, a debos.Action, stage string) int {
+func checkError(context *debos.DebosContext, err error, a debos.Action, stage string) bool {
```

Comment:
Personally I would use with "error", since that gives us the option to bubble it up, plus it seems more consistent with the rest of the codebase.

Reply:
right that actually should be the case for the functions which consume `checkError` too; currnetly they return an int.....

Reply:
Having slept on it, I am leaning towards removing the helper all together. As mentioned previously - it masks the `DebugShell` call, but it also masks the `context.State = debos.Failed` handling. The existing codebase has a bunch of the latter open-coded, making it look fairly imbalanced.

On the other hand the `State` is used solely within the single `PostMachineCleanup`. Perhaps we should check/update all actions adding the callback and/or state-check. Mildly related - we're setting up `defer PostMachineCleanup` prior to the `Pre[No,]Machine` calls, which seems odd.

I'm not sure if these are something to care about (or are even genuine issues). Let me know it they make sense and I can open issues for each one - to be fixed at misc later point.

Reply:
fwiw this helper should really stay; the naming could be better (handleError?) but otherwise it seems ok; 

setting up the defer PostMachineCleanup before any Pre actions is expected such that the cleanups also occur if the Machine setup failed

Reply:
reversed the boolean logic and called it `handleError` - it returns true once an error had been handled.